pyuvdata/uvdata/uvfits.py

```diff
@@ -437,38 +446,62 @@ def read_uvfits(
             self._set_phased()
             # check if we have an spw dimension
             if vis_hdr["NAXIS"] == 7:
-                if vis_hdr["NAXIS5"] > 1:
-                    raise ValueError(
-                        "Sorry.  Files with more than one spectral"
-                        "window (spw) are not yet supported. A "
-                        "great project for the interested student!"
-                    )
-
                 self.Nspws = vis_hdr.pop("NAXIS5")
-
-                self.spw_array = np.int32(uvutils._fits_gethduaxis(vis_hdu, 5)) - 1
+                self.spw_array = uvutils._fits_gethduaxis(vis_hdu, 5).astype(np.int) - 1
 
                 # the axis number for phase center depends on if the spw exists
                 self.phase_center_ra_degrees = np.float(vis_hdr.pop("CRVAL6"))
                 self.phase_center_dec_degrees = np.float(vis_hdr.pop("CRVAL7"))
             else:
                 self.Nspws = 1
-                self.spw_array = np.array([0])
+                self.spw_array = np.array([np.int(0)])
 
                 # the axis number for phase center depends on if the spw exists
                 self.phase_center_ra_degrees = np.float(vis_hdr.pop("CRVAL5"))
                 self.phase_center_dec_degrees = np.float(vis_hdr.pop("CRVAL6"))
 
             # get shapes
-            self.Nfreqs = vis_hdr.pop("NAXIS4")
             self.Npols = vis_hdr.pop("NAXIS3")
             self.Nblts = vis_hdr.pop("GCOUNT")
 
-            self.freq_array = uvutils._fits_gethduaxis(vis_hdu, 4)
-            self.freq_array.shape = (self.Nspws,) + self.freq_array.shape
-            self.channel_width = vis_hdr.pop("CDELT4")
-            self.polarization_array = np.int32(uvutils._fits_gethduaxis(vis_hdu, 3))
+            if self.Nspws > 1:
+                # If this is multi-spw, use the 'flexibile' spectral window setup
+                self._set_flex_spw()
+                uvfits_nchan = vis_hdr.pop("NAXIS4")
+                self.Nfreqs = uvfits_nchan * self.Nspws
+                self.flex_spw_id_array = np.transpose(
+                    np.tile(np.arange(self.Nspws), (uvfits_nchan, 1))
+                ).flatten()
+                fq_hdu = hdu_list[hdunames["AIPS FQ"]]
+                assert self.Nspws == fq_hdu.header["NO_IF"]
+
+                # TODO: This is fine for now, although I (karto) think that this
+                # is relative to the rest_freq, which can be specified as part of
+                # the AIPS SU table.
```

Comment:
I just spent some time reading the uvfits memo and found this:

>The AIPS FQ or frequency table is an essential table unless there is only one spectral window and one frequency setup. It conveys the frequency offsets to be added to the reference frequency as a function of spectral window and frequency setup. The actual observing frequency requires the addition of source-dependent frequency offsets from the AIPS SU table (Section 4.17 on page 34) and time-dependent frequency offsets from the AIPS FO table (Section 4.6 on page 21).

So my read is that the value in the visibility header is the "reference frequency" rather than the "rest frequency" and that if there is an SU table you also need to add in the offsets contained in that table. Does that seem right?

Reply:
It does -- this should have been written as `ref_frequency` (which is actually what Memo 117 calls it). I put in a note about rest frequencies in anticipation of some additional future changes (similar to mentions of SOURCE, in anticipation of multi-source datasets), but then the phrase "rest frequency" clearly got stuck in my brain. Fixed, pushed in 1be5163